File: exercises/java/src/main/java/tax/TaxCalculator.java
package tax;

import java.time.LocalDate;

abstract class TaxCalculator{
    private final int year;

    abstract int calculateTax(Vehicle vehicle);

    TaxCalculator(){
        this(LocalDate.now().getYear());
    }

    TaxCalculator(int year){
        this.year = year;
    }

    int getYear() {
        return year;
    }
}

File: exercises/dotnet/TaxCalculator/TaxCalculator.cs
using System;

namespace TaxCalculator
{
    public abstract class TaxCalculator
    {
        protected int Year { get; }

        public abstract int CalculateTax(Vehicle vehicle);

        protected TaxCalculator() : this(DateTime.Now.Year)
        {

        }

        protected TaxCalculator(int year)
        {
            Year = year;
        }
    }
}
